File: src/features/destination/types/place.ts
import type { Coordinate } from '@/features/map/types/coordinate';

/**
 * Categorias de lugar. `other` só vem da busca externa (TomTom): um shopping,
 * um endereço... `recent` só existe no app: um destino do histórico.
 */
export type PlaceCategory = 'fuel' | 'food' | 'parking' | 'saved' | 'other' | 'recent';

/** Um destino possível. */
export type Place = Coordinate & {
  id: string;
  name: string;
  /** Endereço ou referência curta, exibida sob o nome. */
  address: string;
  category: PlaceCategory;
  /** Marcado pelo usuário. Aparece no filtro "Salvos". */
  saved?: boolean;
};

File: backend/app/schemas/place.py
"""
Lugares da busca de destino. Mesmos campos do tipo `Place` do aplicativo,
para o app usar a resposta direto.
"""

from enum import StrEnum

from pydantic import Field

from app.schemas.base import ApiModel


class PlaceCategory(StrEnum):
    FUEL = "fuel"
    FOOD = "food"
    PARKING = "parking"
    SAVED = "saved"
    # Resultado da busca externa que não é posto, comida nem estacionamento
    # (um shopping, uma faculdade, um endereço). Nunca vem do banco.
    OTHER = "other"


class Place(ApiModel):
    id: str
    name: str
    address: str
    category: PlaceCategory
    saved: bool = False
    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)


class PlaceListResponse(ApiModel):
    places: list[Place]
    # Quantidade de lugares na lista.
    count: int
